force-app/main/default/lwc/marketingCloudCopadoDataTable/marketingCloudCopadoDataTable.js

```diff
@@ -421,10 +423,16 @@ export default class MarketingCloudCopadoDataTable extends LightningElement {
     this.sortData(this.sortedBy, this.sortDirection);
   }
 
-  sortData(fieldname, direction) {
-    
-    let parseData = JSON.parse(JSON.stringify( this.visibleData));
-    // Return the value stored in the field
+  sortData(fieldname, direction) { 
+    let parseData='';
+    // 68 sort the data 
+    if(this.data){
+      parseData = JSON.parse(JSON.stringify(this.data));
+      this.data=null;
+    }else{
+      parseData = JSON.parse(JSON.stringify(this.visibleData));
```

Comment:
where is "visibleData" ever set?

Reply:
inside the short method. at the end after short. it assign  -> this.visibleData = parseData;

 parseData.sort((next, prev) => {
        // console.log('next:',keyValue(next));
        // console.log('prev:',keyValue(prev));
        next = keyValue(next) ? keyValue(next) : ''; // handling null values
        prev = keyValue(prev) ? keyValue(prev) : '';
        // sorting values based on direction
        return isReverse * ((next > prev) - (prev > next));
    });
    this.visibleData = parseData;

Reply:
this is a rather weird setup. lets please rewrite it.
`this.data`, if I understood the rest of the code (not just this snippet) gets the initial value on page load?
then it's reset right afterwards in `sortData()` which instead sets the value to `this.visibleData`

I don't see the benefit of these 2 variables - either please explain with proper code comments what this is doing or reduce it down to one variable.

also, when pasting code, please use the code button:
![image](https://user-images.githubusercontent.com/1917227/193062843-cccd9b32-0876-4670-a543-54e7ac25a09f.png)
`this is code`

Reply:
this.data is using **handleSearch** method , so we can not remove it.
`handleSearch(event) {
---
this.visibleData = this.data.filter((row) => regex.test(row.n) || regex.test(row.t) || regex.test(row.cd) || regex.test(row.cb)
    || regex.test(row.ld) || regex.test(row.lb) || regex.test(row.k));
`